src/compose/ComposeBox.js

```diff
@@ -217,6 +220,47 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  insertAttachment = async (attachments: DocumentPickerResponse[]) => {
+    this.setState(({ numUploading }) => ({
+      numUploading: numUploading + 1,
+    }));
+
+    const { _, auth } = this.props;
+
+    let placeholders = '';
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      placeholders += `${_('[Uploading {fileName}...]()', { fileName })}\n\n`;
+    }
+    this.insertMessageTextAtCursorPosition(placeholders);
+
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment ${i + 1}`;
+      const placeholder = _('[Uploading {fileName}...]()', { fileName });
```

Comment:
Instead of computing this a second time, how about having an array `placeholders`, so that here we can just say `placeholders[i]`?

Then we can still concatenate them for the `insertMessageTextAtCursorPosition` call, say with `placeholders.join('\n\n')`.

The main advantage of computing them just once is that when we're doing it twice, it makes me worry a bit about whether they'll definitely come out the same both times. I'm pretty sure they always do with the code in this PR, but it's the sort of thing where it'd be very easy for them for a future change to the code to accidentally get them out of sync -- so, better to write the code so that structurally it makes sure they're the same.

Reply:
made array of `placeholders` as well as `fileNames`. I also accidentally got the "replacement fileName" (`attachment ${i+1}`) out of sync in this very commit :sweat_smile: (you can see it in the code highlight here) so this makes it a lot more relevant.
btw I fixed that as well and added a corresponding translation string for it.

Reply:
Ha, indeed! And I didn't notice that either :grin: So that's a nice live demonstration of the point.